src/Compiler/Facilities/prim-lexing.fs

```diff
@@ -108,6 +110,24 @@ type StringText(str: string) =
         member _.CopyTo(sourceIndex, destination, destinationIndex, count) =
             str.CopyTo(sourceIndex, destination, destinationIndex, count)
 
+        member this.GetSubTextFromRange(range) =
+            let sourceText = this :> ISourceText
+            let startLine = range.StartLine - 1
+            let line = sourceText.GetLineString startLine
+
+            if range.StartLine = range.EndLine then
+                let length = range.EndColumn - range.StartColumn
```

Comment:
Do we want to have sanity check that EndColumn > StartColumn?

Reply:
Probably yeah